src/main/java/gregtech/api/recipes/ModHandler.java

```diff
@@ -337,6 +345,83 @@ public static Object finalizeIngredient(Object ingredient) {
         return ingredient;
     }
 
+    public static ItemMaterialInfo getRecyclingIngredients(Object... recipe) {
+        Map<Character, Integer> inputCountMap = new HashMap<>();
+        Map<Material, Long> materialStacksExploded = new HashMap<>();
+
+        int itr = 0;
+        while (recipe[itr] instanceof String) {
+            String s = (String) recipe[itr];
+            for (char c : s.toCharArray()) {
+                if (Character.isLowerCase(c)) continue; // skip tools
+                int count = inputCountMap.getOrDefault(c, 0);
+                inputCountMap.put(c, count + 1);
+            }
+            itr++;
+        }
+
+        char lastChar = ' ';
+        for (int i = itr; i < recipe.length; i++) {
+            Object ingredient = recipe[i];
+
+            // Track the current working ingredient symbol
+            if (ingredient instanceof Character) {
+                lastChar = (char) ingredient;
+                continue;
+            }
+
+            // Should never happen if recipe is formatted correctly
+            // In the case that it isn't, this error should be handled
+            // by an earlier method call parsing the recipe.
+            if (lastChar == ' ') return null;
+
+            ItemStack stack;
+            if (ingredient instanceof MetaItem.MetaValueItem) {
+                stack = ((MetaItem<?>.MetaValueItem) ingredient).getStackForm();
+            } else if (ingredient instanceof UnificationEntry) {
+                stack = OreDictUnifier.get((UnificationEntry) ingredient);
+            } else if (ingredient instanceof ItemStack) {
+                stack = (ItemStack) ingredient;
+            } else if (ingredient instanceof Item) {
+                stack = new ItemStack((Item) ingredient, 1);
+            } else if (ingredient instanceof Block) {
+                stack = new ItemStack((Block) ingredient, 1);
+            } else if (ingredient instanceof String) {
+                stack = OreDictUnifier.get((String) ingredient);
+            } else continue; // throw out bad entries
+
+            BiConsumer<MaterialStack, Character> func = (ms, c) -> {
+                long amount = materialStacksExploded.getOrDefault(ms.material, 0L);
+                materialStacksExploded.put(ms.material, (ms.amount * inputCountMap.get(c)) + amount);
+            };
+
+            // First try to get ItemMaterialInfo
+            ItemMaterialInfo info = OreDictUnifier.getMaterialInfo(stack);
+            if (info != null) {
+                for (MaterialStack ms : info.getMaterials()) func.accept(ms, lastChar);
+                continue;
+            }
+
+            // Then try to get a single Material (UnificationEntry needs this, for example)
+            MaterialStack materialStack = OreDictUnifier.getMaterial(stack);
```

Comment:
Why not `info.getMaterial()`? Since `OreDictUnifier.getMaterial(stack)` basically ends up calling this anyway.

Reply:
For whatever reason, some unification data is stored in `materialUnificationInfo`, and others are stored in `stackUnificationInfo`. These two methods of attempting to acquire unification data simply query both options.

We cannot simply call `getMaterial()`, since if it had an `ItemMaterialInfo` with multiple entries, we would only get the first Material back instead of all of them